### PR TITLE
Create UN_Countries_Capitals

### PR DESCRIPTION
**Pull request by MahmoudAlnouri**
Create UN_Countries_Capitals

---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by RamaBio20 — CHANGES_REQUESTED** [review]
creative idea and easy to follow docstring
but I think it should be split into 2 files one in solutions and one in tests

File: solutions/UN_Countries_Capitals [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+}
+
+def get_capital(country):
+    """
+    Returns the capital city of a given country from the UN_COUNTRIES_AND_CAPITALS dictionary.
+    :param country: The name of a UN member country (string).
+    :return: The capital city (string) or None if the country is not in the dictionary.
+    """
+    return UN_COUNTRIES_AND_CAPITALS.get(country.strip())
+'
```
Comment (tamarasaqer) [inline solutions/UN_Countries_Capitals:213]: There’s an unnecessary trailing single quote (') at the end of the function.

**Review by tamarasaqer — COMMENTED** [review]
Hey Mahmoud,

I followed the checklist, and here’s a list of items that were missed in the code:

1. **Test File Name:**  
   - The tests are included in the same file as the function. Ideally, the tests should be in a separate file named `test_get_capital.py`.

2. **Test Class Docstring:**  
   - The test class is missing a docstring to describe its purpose.

3. **Defensive Assertions in the Function:**  
   - The function does not include defensive assertions to handle invalid inputs (e.g., non-string arguments).

4. **Type Annotations:**  
   - The function lacks type annotations for arguments and return values.

5. **Formatting Issue:**  
   - There’s an unnecessary trailing single quote (`'`) at the end of this line:
     ```python
     return UN_COUNTRIES_AND_CAPITALS.get(country.strip())
     ```
     It should be removed to ensure clean and error-free code.

I hope this feedback is helpful, and I wish you the best of luck with your project!

File: solutions/UN_Countries_Capitals [hunk 2]
```diff
@@ -0,0 +1,238 @@
+
+"""
+A module using Dictionary to provide the capital city for some UN Member Countries.
+
+@author: Mahmoud Alnouri
+@date: 9/Jan/2025.
+"""
```
Comment (dadishimwe) [inline solutions/UN_Countries_Capitals:7]: file should be in .py
